src/test/java/org/commcare/formplayer/services/MediaMetaDataServiceTest.java

```diff
@@ -102,39 +112,66 @@ public List<MediaMetadataRecord> answer(InvocationOnMock invocation) throws Thro
 
     @AfterEach
     public void tearDown() {
+        cacheManager.getCache(MEDIA_METADATA_CACHE).clear();
         testFile.delete();
     }
 
     @Test
-    public void testSaveMetaData() {
+    public void testMediaMetadataCacheExists() {
+        Cache cache = cacheManager.getCache(MEDIA_METADATA_CACHE);
+        assertNotNull(cache);
+        assertTrue(cache instanceof CaffeineCache);
+    }
+
+    @Test
+    public void testSaveMetaData_cached() {
+        cacheManager.getCache(MEDIA_METADATA_CACHE).clear();
+
         mediaMetaDataService.saveMediaMetaData(mediaMetaData);
-        Optional<MediaMetadataRecord> fetchedMediaMetaData = mediaMetaDataRepo.findById(mediaMetaData.getId());
-        Assertions.assertTrue(fetchedMediaMetaData.isPresent());
+
+        assertEquals(mediaMetaData, getCachedMetadata(mediaMetaData.getFileId()).get());
+
+        MediaMetadataRecord fetchedMediaMetaData = mediaMetaDataService.findByFileId(mediaMetaData.getFileId());
+        assertEquals(mediaMetaData, fetchedMediaMetaData);
     }
 
     @Test
     public void testDelete() {
         mediaMetaDataService.saveMediaMetaData(mediaMetaData);
-        Optional<MediaMetadataRecord> fetchedMediaMetaData = mediaMetaDataRepo.findById(mediaMetaData.getId());
-        Assertions.assertTrue(fetchedMediaMetaData.isPresent());
         String metadataId = mediaMetaData.getId();
-        mediaMetaDataService.deleteMetaDataById(metadataId);
+        String fileId = mediaMetaData.getFileId();
+        MediaMetadataRecord fetchedMediaMetaData = mediaMetaDataService.findByFileId(fileId);
+        Assertions.assertNotNull(fetchedMediaMetaData);
+        mediaMetaDataService.deleteByFileId(fileId);
         Optional<MediaMetadataRecord> newlyFetchedMediaMetaData = mediaMetaDataRepo.findById(metadataId);
```

Comment:
we should still use service layer here to find the record. We can just do another `findByFileId` call on the service to check for the existing of the record. This will verify that service cache also invalidates when we delete a record.

Reply:
This was a good change to make. I tested it and the cache was not being updated on delete until `@CacheEvict(key  = "#fileId")` annotation was added to `deleteByFileId`.